hw/ip/pwrmgr/doc/_index.md

```diff
@@ -0,0 +1,233 @@
+The fast FSM acknowledges the slow FSM (which made the original power up request) and releases the system reset stage - this enables the processor to begin operation.
+Afterwards, the fast FSM transitions to ACTIVE state and waits for a software low power entry request.
+
+A low power request is initiated by software through a combination of WFI and software low power hint in {{< regref "CONTROL" >}}.
+Specifically, this means if software issues only WFI, the power manager does not treat it as a power down request.
+The notion of WFI is exported from the processor.
+For Ibex, this is currently in the form of core\_sleeping\_o.
+
+In response to the low power entry request, the fast FSM disables all second level clock gating and asserts appropriate resets if required.  The
+fast FSM then requests the slow FSM to take over.
+Once the request is acknowledged, the fast FSM transitions to low power and waits for the next power up request.
+
+
+## Reset Request Handling
+
+There are 3 reset requests in the system - peripheral requested reset such as watchdog, alert escalation reset and non-debug module reset.
+The flash brownout is handled separately and described in [flash handling section](#Flash-Handling) below.
+
+The watchdog and alert escalation resets are handled directly by the power manager, while the non-debug module reset is handled by the reset controller.
+This separation is because the non-debug reset does not affect the life cycle controller, non-volatile storage controllers and alert states.
+There is thus no need to sequence its operation like the others.
+
+The power controller observes reset requests only in two states - the slow FSM LOW\_POWER state and the fast FSM ACTIVE state - these are considered static states of the system.
+When a reset request is received during slow FSM LOW\_POWER state, the system begins its usual power up sequence even if a wakeup has not been received.
+
+When a reset request is received during fast FSM ACTIVE state, either because the system was already active or exited low power due to the reset request,  the fast FSM asserts resets immediately and transitions back through its initialization process.
+Note in this case, the slow FSM is “not activated” and remains in its IDLE state.
+
+
+### Reset Requests Received During Other States
+
+All other states in the slow / fast FSM are considered transition states.
+Resets are not observed in other states because the system will always be transitioning towards one of the steady states (the system is in the process of powering down or powering up).
+Once a steady state is reached, the reset request is then observed and action taken.
+
+### Reset Recording
+
+Of the resets directly handled by the power manager (peripheral reset request, escalation and low power entry), the power manager sequences only 1 at a time (see state diagrams).
```

Comment:
I think this would be easier to read as something like "The power manager only handles one reset request at once, which means it serializes any overlapping reset requests that it must handle directly (peripheral reset request, escalation and low power entry)."

Also, I looked at the state diagrams and I'm not sure how they support the statement. Is the point that there's just one "reset_req" pathway? I'm not sure whether the text needs a pointer to the state diagrams but, if it keeps the pointer, it probably needs to explain why they demonstrate what it's saying.

Reply:
i completely reworded this section. Can you let me know if it's clearer now?

Reply:
Yep, I think I understand now. Thanks!